tests/containers/upload_container_images_to_ecr.pm

```diff
@@ -23,11 +23,18 @@ sub run {
 
     my ($untested_images, $released_images) = get_suse_container_urls();
 
-    my $provider = $self->provider_factory();
-    my $image    = $untested_images->[0];
-    my $tag      = join('-', get_var('PUBLIC_CLOUD_RESOURCE_NAME'), get_current_job_id());
+    my $provider      = $self->provider_factory();
+    $self->{provider} = $provider;
+
+    my $image = $untested_images->[0];
+    my $tag   = join('-', get_var('PUBLIC_CLOUD_RESOURCE_NAME'), get_current_job_id());
     assert_script_run("docker pull $image", 360);
     $provider->push_container_image($image, $tag);
 }
 
+sub post_fail_hook {
```

Comment:
The idea of `publiccloud:basetest()` is to use `cleanup()` when you simply need a function which is called in any case after the tests `run()` finished, see https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/publiccloud/basetest.pm#L81